### PR TITLE
MarketTrend: aggressive default field matching

### PR DESCRIPTION
**Pull request by ryantxu**
The new market trend panel is great once it is setup... but getting all the fields mapped can be challenging.  This PR tries very hard to visualize the data before falling through to the timeseries options.

Out of the box, it will now graph:
image

as
image

The "random walk table" (ie something with value,min,max) will graph as:
image

And if you send a single timeseries to the panel it will give you good up/down indication like:
image

from:
image

### CODE REVIEW

File: public/app/plugins/panel/market-trend/fields.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      const name = getFieldDisplayName(field, frame).toLowerCase();
+      if (info.defaults.includes(name) || info.defaults.includes(field.name)) {
+        return field;
+      }
+    }
+  }
+  return field;
+}
+
+export function prepareCandlestickFields(
```
Comment (torkelo) [inline public/app/plugins/panel/market-trend/fields.ts:92]: This function could need some unit tests

**Review by torkelo — COMMENTED** [review]
This is great! makes it so much easier to use and get started with. 

Did a quick review of the MarketTrendPanel code and wish we could refactor it a bit. Especially the huge useMemo function in with many nested functions is a bit large. Maybe it can be moved out of the render function or broken up and ideally with some tests. 

The inner body of the `<TimeSeries>`  looks identical to TimeSeriesPanel (the zoom, annotation, annotation editor, thresholds editor etc). Maybe we can break that out and share it between TimeSeriesPanel and MarketTrendPanel?

**Comment by ryantxu** [comment]
> looks identical to TimeSeriesPanel

agree -- but can we do that in a new PR.  I am extra nervous about anythign touching timeseries panel!

**Comment by ryantxu** [comment]
@torkelo -- can you take another look?  I added tests and a little cleanup, but would like to attempt more in a new PR (maybe after resolving naming issues too -- see #41582)

**Review by torkelo — APPROVED** [review]
Nice!

**Pull request merged**